hana/lib/drivers/base.js

```diff
@@ -37,20 +37,58 @@ class HANADriver {
         if (streams.length && changes === 0) {
           changes = 1
         }
+
+        if (streams.length) {
+          Promise.all(
+            streams.map(
+              stream =>
+                new Promise((resolve, reject) => {
+                  stream.on('end', () => {
+                    resolve()
+                  })
+                  stream.on('error', () => {
+                    // REVISIT: does it actually matter if it fails or is successful, so should we resolve here?
+                    reject()
```

Comment:
As the comment states this should resolve. The main reason is that it is a dangling promise. Which means if you `reject` it will go to the root of the process will crash the application.